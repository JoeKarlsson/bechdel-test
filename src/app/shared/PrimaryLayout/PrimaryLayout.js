import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Header from '../Header/Header';
import Footer from '../Footer/Footer';
import About from '../../static/About/About';
import CaseStudy from '../../static/CaseStudy/CaseStudy';
import Films from '../../films/Films/Films';
import Film from '../../films/Film/Film';
import NewFilm from '../../films/NewFilm/NewFilm';
import NoMatch from '../../static/NoMatch/NoMatch';
import ErrorBoundary from '../ErrorBoundary/ErrorBoundary';
import '../../assets/styles/skeleton.css';
import '../../assets/styles/normalize.css';
import './PrimaryLayout.scss';

export const PrimaryLayout = () => {
	return (
		<div className="PrimaryLayout">
			<Header />
			<div className="content">
				<div className="container">
					<ErrorBoundary>
						<Switch>
							<Route exact path="/" component={Films} />
							<Route path="/about" component={About} />
							<Route path="/case-study" component={CaseStudy} />
							<Route path="/film/new" component={NewFilm} />
							<Route path="/film/:id" component={Film} />
							<Route path="*" component={NoMatch} />
						</Switch>
					</ErrorBoundary>
				</div>
			</div>
			<Footer />
		</div>
	);
};

export default PrimaryLayout;
