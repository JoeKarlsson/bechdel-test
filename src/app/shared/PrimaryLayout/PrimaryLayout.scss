a {
	text-decoration: none;
}

.PrimaryLayout {
	display: flex;
	min-height: 100vh;
	flex-direction: column;
}

.content {
	// padding-top: 2%;
	// padding-bottom: 2%;
	flex: 1;
}
