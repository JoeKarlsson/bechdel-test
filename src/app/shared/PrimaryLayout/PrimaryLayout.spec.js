import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import PrimaryLayout from './PrimaryLayout';

jest.mock('../../films/NewFilm/Uploader/Uploader');
jest.mock('../../helper/api');

configure({ adapter: new Adapter() });

jest.mock('../../films/NewFilm/Uploader/Uploader');

describe('PrimaryLayout Page', () => {
	let wrapper;
	let inst;

	beforeEach(() => {
		wrapper = shallow(
			<MemoryRouter>
				<PrimaryLayout />
			</MemoryRouter>
		);
		inst = wrapper.instance();
	});

	describe('rendering', () => {
		describe('initial state', () => {
			it('should match the snapshot', () => {
				const component = renderer.create(
					<MemoryRouter>
						<PrimaryLayout />
					</MemoryRouter>
				);
				const tree = component.toJSON();
				expect(tree).toMatchSnapshot();
			});

			it('is rendered correctly', () => {
				expect(wrapper).toHaveLength(1);
			});

			it('should render without throwing an error', () => {
				expect(wrapper.text()).toEqual('<Router />');
			});

			it('should be selectable by the class `PrimaryLayout`', () => {
				expect(
					wrapper
						.dive()
						.dive()
						.is('.PrimaryLayout')
				).toBe(true);
			});

			it('should mount in the full DOM', () => {
				expect(
					wrapper
						.dive()
						.dive()
						.find('.PrimaryLayout').length
				).toBe(1);
			});

			it('should render to static HTML', () => {
				expect(
					wrapper
						.dive()
						.dive()
						.text()
				).toContain('<Header /><ErrorBoundary /><Footer />');
			});

			it('should have correct inital state', () => {
				const initialState = inst.state;
				const expectedIntialState = {};
				expect(initialState).toMatchObject(expectedIntialState);
			});

			it('should not have any inital props', () => {
				const initialProps = inst.props;
				const expectedProps = {
					children: <PrimaryLayout />,
				};
				expect(initialProps).toEqual(expectedProps);
			});
		});
	});
});
