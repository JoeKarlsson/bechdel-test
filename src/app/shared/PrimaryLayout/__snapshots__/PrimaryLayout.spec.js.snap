// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryLayout Page rendering initial state should match the snapshot 1`] = `
<div
  className="PrimaryLayout"
>
  <header
    className="header_bar"
  >
    <a
      aria-current="true"
      className="header_logo active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      bechdel.io
    </a>
    <ul
      className="header_nav"
    >
      <li>
        <a
          aria-current={false}
          className={undefined}
          href="/about"
          onClick={[Function]}
          style={undefined}
        >
          about
        </a>
      </li>
      <li>
        <a
          aria-current={false}
          className={undefined}
          href="/case-study"
          onClick={[Function]}
          style={undefined}
        >
          case study
        </a>
      </li>
    </ul>
  </header>
  <div
    className="content"
  >
    <div
      className="Home"
    >
      <div
        className="section hero"
      >
        <div
          className="container"
        >
          <div
            className="row"
          >
            <div
              className="u-max-full-width "
            >
              <h4
                className="hero-heading"
              >
                bechdel.io
              </h4>
              <h5>
                find out if your favorite film passes the test.
              </h5>
              <a
                aria-current={false}
                className={undefined}
                href="/film/new"
                onClick={[Function]}
                style={undefined}
              >
                <button
                  className="button"
                >
                  upload script
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
      <div>
        Loading...
      </div>
    </div>
  </div>
  <footer
    className="footer"
  >
    <div
      className="footer_content"
    >
      <a
        href="https://github.com/JoeKarlsson/bechdel-test"
        rel="noopener noreferrer"
        target="_blank"
      >
        <p>
          2018
           - a karlsson production.
        </p>
      </a>
    </div>
  </footer>
</div>
`;
